### PR TITLE
Coverage DurationFormat formatToParts

### PR DESCRIPTION
**Pull request by romulocintra**
Test formatToParts output https://github.com/tc39/proposal-intl-duration-format/pull/126

### CODE REVIEW

**Comment by romulocintra** [comment]
cc @sffc

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    { type: "literal", value: " " },
+    { type: "integer", value: "789", unit: "nanosecond" },
+    { type: "unit", value: "ns", unit: "nanosecond" },
+  ],
+  digital: [
+    { type: "integer", value: "7", unit: "hour" },
+    { type: "literal", value: ":", unit: "hour" },
+    { type: "integer", value: "08", unit: "minute" },
+    { type: "literal", value: ":", unit: "minute" },
+    { type: "integer", value: "09", unit: "second" },
```
Comment (ryzokuken) [inline test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:117]: I think you missed the `.123456789`

**Review by ryzokuken — APPROVED** [review]
Overall LGTM. Thanks!

**Review by sffc — APPROVED** [review]
:+1: once @ryzokuken's comment is addressed

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js (hunk 1 above)
Comment (romulocintra) [reply test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:117]: Hi, I have a doubt if by default we do
```
8. If listStyle is "digital", then
      a. Set listStyle to "short".
```

To see the sub-second values we shouldn't define the number of fractionalDigits ?       

```js

new Intl.DurationFormat('en',{ style : 'digital' , fractionalDigits: 9 }).formatToParts( {
  hours: 7,
  minutes: 8,
  seconds: 9,
  milliseconds: 123,
  microseconds: 456,
  nanoseconds: 789,
});

// Output  something like 

    { type: "integer", value: "7", unit: "hour" },
    { type: "literal", value: ":", unit: "hour" },
    { type: "integer", value: "08", unit: "minute" },
    { type: "literal", value: ":", unit: "minute" },
    { type: "integer", value: "09", unit: "second" },
    { type: "literal", value: ".", unit: "second" },
    { type: "integer", value: "123456789", unit: "millisecond" },


```

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  assert.sameValue(Array.isArray(expected), true, `${message}: expected is Array`);
+  assert.sameValue(Array.isArray(actual), true, `${message}: actual is Array`);
+  assert.sameValue(actual.length, expected.length, `${message}: length`);
+
+  for (const [i, actualEntry, expectedEntry] of zip(actual, expected)) {
+    // assertions
+    assert.sameValue(actualEntry.type, expectedEntry.type, `type for entry ${i}`);
+    assert.sameValue(actualEntry.value, expectedEntry.value, `value for entry ${i}`);
+    if(expectedEntry.unit){
+      assert.sameValue(actualEntry.unit, expectedEntry.unit, `source for entry ${i}`);
```
Comment (ptomato) [inline test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:29]:
```suggestion
      assert.sameValue(actualEntry.unit, expectedEntry.unit, `unit for entry ${i}`);
```

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+// Utils functions
+function* zip(a, b) {
+  assert.sameValue(a.length, b.length);
+  for (let i = 0; i < a.length; ++i) {
+    yield [i, a[i], b[i]];
+  }
+}
+
+function compare(actual, expected) {
```
Comment (ptomato) [inline test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:18]:
```suggestion
function compare(actual, expected, message) {
```

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+/*---
+esid: sec-Intl.DurationFormat.prototype.formatToParts
+description:  Checks basic handling of formatToParts, using long, short,narrow and digital styles.
+features: [Intl.DurationFormat]
+---*/
+
+// Utils functions
+function* zip(a, b) {
+  assert.sameValue(a.length, b.length);
```
Comment (ptomato) [inline test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:12]: This is already asserted below, so I think we can remove it from here.

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js (hunk 1 above)
Comment (ptomato) [reply test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:117]: I think this is done in [PartitionDurationFormatPattern](https://tc39.es/proposal-intl-duration-format/#sec-partitiondurationformatpattern) step 4.j.iv.

The default `fractionalDigits` is 0, so I think this test is correct.

**Review by ptomato — APPROVED** [review]
A few nitpicks about the assertions but otherwise looks good. I agree the test is correct specifically on the question that Ujjwal raised.

File: test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js (hunk 1 above)
Comment (ryzokuken) [reply test/intl402/DurationFormat/prototype/formatToParts/formatToParts-styles-en.js:117]: You're right! Sorry for the false alarm.

**Pull request merged**